03_scorep/scorep_notes.md

```diff
@@ -0,0 +1,69 @@
+# Score-P with BOUT++/STORM
+
+See workshop slides for intro.
+
+Score-P configuration with BOUT++ was set up with a previous project (with EPCC?). This
+configuration only instruments MPI calls, because fully instrumenting the code makes runs
+unmanageably slow. To fully instrument the code, edit `make.config` to remove the
+`--nocompiler` flag. Then need to filter...
+
+### Filtering
+
+Filtering is needed both to prevent Score-P from running out of memory to store its
+profiling information, and to prevent excessive overhead from instrumentation making the
+profiling uninformative.
+
+Running with a short enough `timestep` and `nout` with full instrumentation (a few hundred
+rhs calls should be plenty) should let Score-P produce some output.
+
+> Without filtering the runs may be very slow (observed with icc/intelmpi on DINE).
+> However on Marconi with gcc/openmpi the unfiltered run took only about 9 mins, compared
+> to 1 min 50 s for a plain, non-Score-P run.
```

Comment:
icc/intelmpi 2020 update 2 (the intel toolchain available for scorep) on DINE is a bit non-functional.  I tried these runs around the same time that _all_ my runs stopped behaving normally.